test/controllers/update/overseas.entity.presenter.controller.spec.ts

```diff
@@ -0,0 +1,174 @@
+
+const mockSetApplicationData = setApplicationData as jest.Mock;
+const mockAuthenticationMiddleware = authentication as jest.Mock;
+mockAuthenticationMiddleware.mockImplementation((req: Request, res: Response, next: NextFunction) => next() );
+
+const mockPrepareData = prepareData as jest.Mock;
+
+describe("PRESENTER controller", () => {
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    mockSetApplicationData.mockReset();
+  });
+
+  describe("GET tests", () => {
+    test(`renders the presenter page`, async () => {
+      mockGetApplicationData.mockReturnValueOnce({ [PresenterKey]: PRESENTER_OBJECT_MOCK, [OeNumberKey]: "OE123456" });
+      const resp = await request(app).get(OVERSEAS_ENTITY_PRESENTER_URL);
+
+      expect(resp.status).toEqual(200);
+      expect(resp.text).toContain(LANDING_PAGE_URL);
+      expect(resp.text).toContain(OVERSEAS_ENTITY_PRESENTER_PAGE_TITLE);
+      expect(resp.text).not.toContain(PAGE_TITLE_ERROR);
+      expect(resp.text).toContain(UPDATE_USE_INFORMATION_NEED_MORE);
+      expect(resp.text).toContain(INFORMATION_SHOWN_ON_THE_PUBLIC_REGISTER);
+      expect(resp.text).toContain(NOT_SHOW_INFORMATION_ON_PUBLIC_REGISTER);
+    });
+
+    test("catch error when renders the presenter page", async () => {
+      mockGetApplicationData.mockImplementationOnce( () => { throw new Error(ANY_MESSAGE_ERROR); });
+      const resp = await request(app).get(OVERSEAS_ENTITY_PRESENTER_URL);
+
+      expect(resp.status).toEqual(500);
+      expect(resp.text).toContain(SERVICE_UNAVAILABLE);
+    });
+  });
+
+  describe("POST tests", () => {
+    test(`redirect to the ${OVERSEAS_ENTITY_REVIEW_URL} page after a successful post from presenter page`, async () => {
+      const resp = await request(app).post(OVERSEAS_ENTITY_PRESENTER_URL).send(PRESENTER_OBJECT_MOCK);
+
+      expect(resp.status).toEqual(302);
+      expect(resp.text).toContain(`${FOUND_REDIRECT_TO} ${OVERSEAS_ENTITY_REVIEW_URL}`);
+    });
+
+    test(`redirect to the ${OVERSEAS_ENTITY_REVIEW_URL} page after a successful post from presenter page with special characters`, async () => {
+      const resp = await request(app).post(OVERSEAS_ENTITY_PRESENTER_URL).send(PRESENTER_WITH_SPECIAL_CHARACTERS_FIELDS_MOCK);
+
+      expect(resp.status).toEqual(302);
+      expect(resp.text).toContain(`${FOUND_REDIRECT_TO} ${OVERSEAS_ENTITY_REVIEW_URL}`);
+    });
+
+    test("renders the current page with error message", async () => {
+      const resp = await request(app).post(OVERSEAS_ENTITY_PRESENTER_URL).send({ email: '' });
+
+      expect(resp.status).toEqual(200);
+      expect(resp.text).toContain(OVERSEAS_ENTITY_PRESENTER_PAGE_TITLE);
+      expect(resp.text).toContain(ErrorMessages.FULL_NAME);
+      expect(resp.text).toContain(ErrorMessages.EMAIL);
+      expect(resp.text).not.toContain(ErrorMessages.MAX_EMAIL_LENGTH);
+      expect(resp.text).not.toContain(ErrorMessages.EMAIL_INVALID_FORMAT);
+      expect(resp.text).toContain(LANDING_URL);
```

Comment:
contains LANDING_URL?

Reply:
@AndreKramer7654 , the previous logic used to send the user to the page that was the landing page, as the landing page for update might change this will redirect the user to the currently defined landing page

Reply:
But not to the landing page for update? Would it not be better to test for that?

Reply:
oh yes, good spot...

Reply:
Removed the check as it was testing the SERVICE_NAME header contained the landing page url and we have still not implemented the change to modify SERVICE_NAME